impl/sql/mutationstorage/queue.go

```diff
@@ -61,29 +61,44 @@ func (m *Mutations) Send(ctx context.Context, domainID string, update *pb.EntryU
 	return m.send(ctx, domainID, logID, mData, time.Now())
 }
 
-// randLog returns a random, enabled log for domainID.
-func (m *Mutations) randLog(ctx context.Context, domainID string) (int64, error) {
-	// TODO(gbelvin): Cache these results.
+// ListLogs returns a list of all logs for domainID, optionally filtered for writable logs.
+func (m *Mutations) ListLogs(ctx context.Context, domainID string, writable bool) ([]int64, error) {
+	var query string
+	if writable {
+		query = `SELECT LogID from Logs WHERE DomainID = ? AND Enabled = True;`
+	} else {
+		query = `SELECT LogID from Logs WHERE DomainID = ?;`
+	}
 	var logIDs []int64
-	rows, err := m.db.QueryContext(ctx,
-		`SELECT LogID from Logs WHERE DomainID = ? AND Enabled = ?;`,
-		domainID, true)
+	rows, err := m.db.QueryContext(ctx, query, domainID)
 	if err != nil {
-		return 0, err
+		return nil, err
 	}
+
 	defer rows.Close()
 	for rows.Next() {
 		var logID int64
 		if err := rows.Scan(&logID); err != nil {
-			return 0, err
+			return nil, err
 		}
 		logIDs = append(logIDs, logID)
 	}
 	if err := rows.Err(); err != nil {
-		return 0, err
+		return nil, err
 	}
 	if len(logIDs) == 0 {
-		return 0, status.Errorf(codes.NotFound, "no log found for domain %v", domainID)
+		return nil, status.Errorf(codes.NotFound, "no log found for domain %v", domainID)
+	}
+	return logIDs, nil
+}
+
+// randLog returns a random, enabled log for domainID.
+func (m *Mutations) randLog(ctx context.Context, domainID string) (int64, error) {
```

Comment:
Do you use this function anywhere? I see it only in tests.

Reply:
`randLog` is used by `Send`